### PR TITLE
Add BloomFilter

### LINKED ISSUE #8: Implement a BloomFilter
A `Filter` trait has been added without an implementation.
We can add a `BloomFilter` like LevelDB.

Reference: https://github.com/google/leveldb/blob/master/util/bloom.cc

### PR DESCRIPTION
**Pull request by ZhongliGao**
close #8

### CODE REVIEW

**Comment by ZhongliGao** [comment]
close #8

**Review by huachaohuang — COMMENTED** [review]
Would you please add some comments to explain the algorithm? I don't really understand it at all.

File: engula/src/format/bloom_filter.rs [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+}
+
+impl BloomFilter {
+    #[allow(dead_code)]
+    // count : number of elements to insert into bloom filter
+    // false_positive : Bloom filter false_positive rate
+    pub fn new(count: usize, false_positive: f64) -> BloomFilter {
+        let m = m_size(count, false_positive);
+        let k = k_size(false_positive);
+        let hashers = [
```
Comment (huachaohuang) [inline engula/src/format/bloom_filter.rs:32]: Is that OK to use the same hash function here?

File: engula/src/format/bloom_filter.rs [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+        let hash_1 = hasher_1.finish();
+        let hash_2 = hasher_2.finish();
+
+        (hash_1, hash_2)
+    }
+
+    // Get the index from hash value of `k_i`.
+    fn get_index(&self, h1: u64, h2: u64, i: u64) -> usize {
+        (h1.wrapping_add((i).wrapping_mul(h2)) % self.m) as usize
```
Comment (huachaohuang) [inline engula/src/format/bloom_filter.rs:80]: I think LevelDB's implementation is more efficient. Would you like to optimize it?

**Review by huachaohuang — CHANGES_REQUESTED** [review]
Nice try :)

**Review by huachaohuang — APPROVED** [review]
OK, I can merge it now, but I may refactor it later.

**Pull request merged**